pkg/volunteer/volunteer.go

```diff
@@ -103,12 +109,18 @@ func (v *volunteer) generateRecord() (report.Record, error) {
 		return report.Record{}, err
 	}
 
+	extensions, err := v.extensionsLister.ListExtensions()
+	if err != nil {
+		return report.Record{}, err
+	}
```

Comment:
IMO this may be the only contentious part of this PR. I can imagine a corrupt or non-existent file (due to some external file generation errors) could cause the volunteer to never send _any_ stats. 

Instead of hard failing in those cases it would arguable be better to either:
1. just omit the extensions altogether and proceed with sending the payload
2. send a special extension key/value pair indicating an error occurred

I would lean towards 1 to avoid sending unexpected fields or adding too much junk data to sift through.